packages/@aws-cdk/aws-cognito/lib/user-pool.ts

```diff
@@ -365,6 +378,148 @@ export interface PasswordPolicy {
+  readonly mediumAction?: AccountTakeoverAction;
+}
+
+/**
+ * The configuration for actions and notification for different levels of risk detected for a potential account takeover.
+ */
+export interface AccountTakeoverRiskConfiguration {
+  /**
+   * The actions of account takeover risk configuration.
+   */
+  readonly actions: AccountTakeoverActions;
+
+  /**
+   * The notify configuration used to construct email notifications.
+   * @default - no configured
+   */
+  readonly notifyConfiguration?: CfnUserPoolRiskConfigurationAttachment.NotifyConfigurationTypeProperty;
+}
+
+/**
+ * The action for compromised credentials.
+ * @see https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-pool-settings-compromised-credentials.html
+ */
+export enum CompromisedCredentialsEventAction {
+  /** Blocking requires users to choose another password */
+  BLOCK = 'BLOCK',
+  /** all attempted uses of compromised credentials to Amazon CloudWatch */
+  NO_ACTION = 'NO_ACTION',
+}
+
+/**
+ * The actions for compromised credentials.
+ */
+export interface CompromisedCredentialsActions {
+  /**
+   * The action compromised credentials event.
+   */
+  readonly eventAction: CompromisedCredentialsEventAction;
+}
+
+/**
+ * The event type for compromised credentials check.
+ * @see https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-pool-settings-compromised-credentials.html
+ */
+export enum CompromisedCredentialsEvent {
+  /** Users sign in */
+  SIGN_IN = 'SIGN_IN',
+  /** Users change password */
+  PASSWORD_CHANGE = 'PASSWORD_CHANGE',
+  /** Users sign up */
+  SIGN_UP = 'SIGN_UP',
+}
+
+/**
+ * The configuration for compromised credentials risk.
+ */
+export interface CompromisedCredentialsRiskConfiguration {
+  /**
+   * The actions for compromised credentials risk.
+   */
+  readonly actions: CompromisedCredentialsActions;
+  /**
+   * The trigger for compromised credentials check.
+   * @default - Not set.
+   */
+  readonly eventFilter?: CompromisedCredentialsEvent[];
+}
+
+/**
+ * Risk settings for advanced security.
+ */
+export interface UserPoolRiskConfigurationOptions {
+  /**
+   * The identifier of the app client to which you want to specify the risk setting.
+   * You can specify a single client or all clients.
+   */
+  readonly clientId: string;
+
+  /**
+   * The account takeover risk configuration including the notify configuration and actions.
+   * @default - no configured
+   */
+  readonly accountTakeoverRiskConfiguration?: AccountTakeoverRiskConfiguration;
+  /**
+   * The compromised credentials risk configuration including the event filter and action.
+   * @default - no configured
+   */
+  readonly compromisedCredentialsRiskConfiguration?: CompromisedCredentialsRiskConfiguration;
+  /**
+   * The configuration to override the risk decision.
+   * @default - no configured
+   */
+  readonly riskExceptionConfiguration?: CfnUserPoolRiskConfigurationAttachment.RiskExceptionConfigurationTypeProperty;
```

Comment:
We don't expose the underlying Cfn resources here. Hide this under a custom modeled struct.